frontend/src/components/roster/View.js

```diff
@@ -0,0 +1,355 @@
+import {
+  createMutation,
+  createQuery,
+  useQueryClient
+} from "@tanstack/solid-query";
+import clsx from "clsx";
+import { trophy } from "solid-heroicons/solid";
+import { For, Match, Show, Switch } from "solid-js";
+
+import {
+  fetchTeamBySlug,
+  fetchTournamentBySlug,
+  fetchTournamentTeamBySlug,
+  removeFromRoster
+} from "../../queries";
+import { useStore } from "../../store";
+import Warning from "../alerts/Warning";
+import Breadcrumbs from "../Breadcrumbs";
+import EditRosteredPlayer from "./EditRosteredPlayer";
+import RemoveFromRoster from "./RemoveFromRoster";
+
+const Roster = () => {
+  const queryClient = useQueryClient();
+  const params = useParams();
+  const [store] = useStore();
+
+  const tournamentQuery = createQuery(
+    () => ["tournament", params.tournament_slug],
+    () => fetchTournamentBySlug(params.tournament_slug)
+  );
+
+  const teamQuery = createQuery(
+    () => ["team", params.team_slug],
+    () => fetchTeamBySlug(params.team_slug)
+  );
+
+  const rosterQuery = createQuery(
+    () => ["tournament-roster", params.tournament_slug, params.team_slug],
+    () => fetchTournamentTeamBySlug(params.tournament_slug, params.team_slug)
+  );
+
+  const removeFromRosterMutation = createMutation({
+    mutationFn: removeFromRoster,
+    onSuccess: () =>
+      queryClient.invalidateQueries({ queryKey: ["tournament-roster"] })
+  });
+
+  const currentUserIsTeamAdmin = () =>
+    teamQuery.data?.admins.filter(user => user.id === store.data.id).length ==
+    1;
+
+  const isPlayer = registration => registration?.roles?.includes("player");
+
+  const isCaptain = registration =>
+    registration?.roles?.includes("captain") ||
+    registration?.roles?.includes("Captain");
+
+  const isSpiritCaptain = registration =>
+    registration?.roles?.includes("spirit captain") ||
+    registration?.roles?.includes("Spirit Captain");
+
+  const isCoach = registration =>
+    registration?.roles?.includes("coach") ||
+    registration?.roles?.includes("Coach") ||
+    registration?.roles?.includes("assistant coach") ||
+    registration?.roles?.includes("Assistant Coach");
+
+  const isManager = registration => registration?.roles?.includes("Manager");
+
+  const players = () => rosterQuery.data?.filter(reg => isPlayer(reg));
+  const nonPlayers = () => rosterQuery.data?.filter(reg => !isPlayer(reg));
+
+  return (
+    <Show
+      when={!tournamentQuery.data?.message}
+      fallback={
+        <div>
+          Tournament could not be fetched. Error -{" "}
+          {tournamentQuery.data.message}
+          <A href={"/tournaments"} class="text-blue-600 dark:text-blue-500">
+            <br />
+            Back to Tournaments Page
+          </A>
+        </div>
+      }
+    >
+      <Breadcrumbs
+        icon={trophy}
+        pageList={[
+          { url: "/tournaments", name: "All Tournaments" },
+          {
+            url:
+              tournamentQuery.data?.status === "REG"
```

Comment:
this page can always be entered only from reg place.. so shoudnt we always take them back to registration page only?